### ISSUE TITLE
MQTT Abnormal State Error shown when not in error

### ISSUE THREAD

**Issue by Jchuchla** [FalconChristmas/fpp#1160]
# Please answer the following:
**FPP Version and Hardware**: 
v5.3.1 (v5.3 branch) On Pi 3b

Status Page Red Bar Error shown:
Abnormal Conditions - May Cause Poor Performance
MQTT Disconnected

Checked MQTT broker and FPP is indeed sending updates

**Expected behavior**: 
Abnormal conditions error message should clear when MQTT reconnects.
Message should be dismissible

**Additional Attachments**
Logs can be provided upon request.

**Comment by ghormann** [comment]
Pleas attach the logs, ideally with control in debug mode.

**Comment by Jchuchla** [comment]
The error message cleared when I restarted FPPD.  I'm able to do that now, but don't want to have to restart FPPD while the show season is in full swing just to clear an MQTT error.
Do you need control in debug mode when the error starts, or can i turn it on now and you'll see what you need to see?
Which logs do you need? Here's the logs I've got right now
[FPP_Logs_20211121.0038.zip](https://github.com/FalconChristmas/fpp/files/7575589/FPP_Logs_20211121.0038.zip)

**Comment by Jchuchla** [comment]
I did have a power outage earlier in the week.  I'm sure my MQTT broker is much slower to boot than FPP.  The broker is running in a windows VM on HyperV.  So it's possible that the message was generated when FPP booted and couldn't connect.  Nonetheless, it did eventually connect.  The message should automatically go away when it does connect successfully.

**Comment by ghormann** [comment]
Control would have to be in debug mode prior to the error.  Though the fact you think FPP booted before MQTT is a helpful hint.

**Comment by Jchuchla** [comment]
I also have another ticket related to FPP not receiving multicast universe input after a whole house power failure.  In that case, I think it’s because the FPP tries to register the IGMP subscriptions before the switch is up and never tries again till FPPD is restarted.  It’s probably not related to MQTT, but it looks like it’s probably the same cause as this issue.

In my environment, I use enterprise network switches.  And my DHCP, DNS and NTP services run on a windows server VM.  My main switch takes a good 6 minutes to boot.  The hyper-V host takes about 10-12 minutes to boot, and then another 7-10 minutes for the VM to become ready.  The FPP instances  boot way faster than the network switches.

NTP on the FPP always catches up as soon as the server becomes available.  The MQTT should do the same.  Actually it seems like it does actually connect.  But the error message should go away as well.

**Comment by patdelaney** [comment]
can't reproduce

**Comment by darylc** [comment]
FPP team cannot reproduce

**Comment by darylc** [comment]
Dan suggests it may be fixed in 7.2 or later

**Comment by Jchuchla** [comment]
Doesn't seem to be fixed.  I just had this issue happen again today during a snowstorm power failure.  The FPP is currently on 7x-417 master branch.  That should be newer than 7.3.

**Comment by nmbgeek** [comment]
I am on 8.2 and seeing a similar issue.  My network is connected through a wireless bridge which is on same power source as my show and FPP ends up starting before the network is ready.  Restarting FFPD with network up and running does resolve the issue.  Since the PI doesn't have an RTC I am assuming that is why the date/time is off on startup.

```
2024-09-06 04:26:48.791 (1323) [General] /opt/fpp/src/log.cpp:353: =========================================
2024-09-06 04:26:48.792 (1323) [General] /opt/fpp/src/log.cpp:354: FPP 8.2
2024-09-06 04:26:48.792 (1323) [General] /opt/fpp/src/log.cpp:355: Branch: v8.2
2024-09-06 04:26:48.793 (1323) [General] /opt/fpp/src/log.cpp:367: Model: Raspberry Pi 4 Model B Rev 1.2
2024-09-06 04:26:48.793 (1323) [General] /opt/fpp/src/log.cpp:371: =========================================
2024-09-06 04:26:48.795 (1323) [General] /opt/fpp/src/fppd.cpp:661: Creating MQTT
2024-09-06 04:26:48.795 (1323) [Control] /opt/fpp/src/mqtt.cpp:175: Using MQTT client id of FPP-Main_8957 
2024-09-06 04:26:48.795 (1323) [Control] /opt/fpp/src/mqtt.cpp:198: No CA File specified for MQTT
2024-09-06 04:26:48.795 (1323) [Control] /opt/fpp/src/mqtt.cpp:214: Error, unable to connect to Mosquitto Broker at 10.0.0.243: 14
2024-09-06 04:26:48.795 (1323) [Control] /opt/fpp/src/mqtt.cpp:216: MQTT Error: Bad address
2024-09-06 04:26:48.795 (1323) [Control] /opt/fpp/src/fppd.cpp:665: MQTT Init failed. Starting without MQTT. -- Maybe MQTT host doesn't resolve
2024-09-06 04:26:48.796 (1323) [General] /opt/fpp/src/fppd.cpp:709: Creating Scheduler, Playlist, and Sequence
2024-09-06 04:26:48.811 (1323) [Schedule] /opt/fpp/src/Scheduler.cpp:729: Current Schedule: (Status: '+' = Enabled, '-' = Disabled, '!' = Outside Date Range, '*' = Repeat, Stop (G)raceful/(L)oop/(H)ard
2024-09-06 04:26:48.811 (1323) [Schedule] /opt/fpp/src/Scheduler.cpp:730: Stat Start & End Dates       Days          Start & End Times   Playlist/Command
2024-09-06 04:26:48.811 (1323) [Schedule] /opt/fpp/src/Scheduler.cpp:731: ---- ----------------------- ------------- ------------------- ---------------------------------------------
2024-09-06 04:26:48.811 (1323) [Schedule] /opt/fpp/src/Scheduler.cpp:734: + *G 2022-11-25 - 2028-12-31 Everyday      SunSet   - 22:00:00 Christmas_Music
2024-09-06 04:26:48.811 (1323) [Schedule] /opt/fpp/src/Scheduler.cpp:734: - *G 2022-12-07 - 2027-12-31 Everyday      22:01:00 - 22:10:00 Christmas Countdown
2024-09-06 04:26:48.811 (1323) [Schedule] /opt/fpp/src/Scheduler.cpp:734: +  G 2023-12-06 - 2028-12-31 Everyday      22:00:00            Remote Falcon - Turn Viewer Control Off
2024-09-06 04:26:48.811 (1323) [Schedule] /opt/fpp/src/Scheduler.cpp:734: +  G 2022-11-25 - 2028-12-31 Everyday      SunSet              Remote Falcon - Update Remote Playlist
2024-09-06 04:26:48.811 (1323) [Schedule] /opt/fpp/src/Scheduler.cpp:734: +  G 2022-11-25 - 2028-12-31 Everyday      SunSet              Remote Falcon - Turn Viewer Control On
2024-09-06 04:26:48.827 (1323) [General] /opt/fpp/src/fppd.cpp:712: Creation of Scheduler, Playlist, and Sequence Complete
2024-09-06 04:26:49.898 (1342) [ChannelOut] /opt/fpp/src/channeloutput/UDPOutput.cpp:681: Could not ping host 10.0.0.205, removing from output
2024-09-06 04:26:49.903 (1342) [ChannelOut] /opt/fpp/src/channeloutput/UDPOutput.cpp:681: Could not ping host 10.0.0.206, removing from output
2024-09-06 04:26:50.260 (1323) [ChannelOut] /opt/fpp/src/channeloutput/ChannelOutputSetup.cpp:321: UDPOutput:  Determined range needed 0 - 1530
2024-09-06 04:26:50.260 (1323) [ChannelOut] /opt/fpp/src/channeloutput/ChannelOutputSetup.cpp:321: UDPOutput:  Determined range needed 1530 - 27030
2024-09-06 04:26:50.260 (1323) [ChannelOut] /opt/fpp/src/channeloutput/ChannelOutputSetup.cpp:351: Determined range needed 0 - 27030
2024-09-06 04:26:50.265 (1323) [Command] /opt/fpp/src/command.cpp:70: Initializing Command Module
2024-09-06 04:26:50.269 (1323) [E131Bridge] /opt/fpp/src/e131bridge.cpp:316: Universe Count = 0
2024-09-06 04:26:50.373 (1323) [General] /opt/fpp/src/fppd.cpp:894: Starting main processing loop
2024-09-06 04:26:53.373 (1388) [Control] /opt/fpp/src/Events.cpp:218: Starting Publish Thread
2024-09-06 04:28:50.452 (1323) [ChannelOut] /opt/fpp/src/channeloutput/UDPOutput.cpp:356: UDP Interface eth0 now down
2024-09-06 04:28:50.453 (1342) [ChannelOut] /opt/fpp/src/channeloutput/UDPOutput.cpp:654: Could ping host 10.0.0.206, re-adding to outputs
2024-09-06 04:28:50.459 (1342) [ChannelOut] /opt/fpp/src/channeloutput/UDPOutput.cpp:654: Could ping host 10.0.0.205, re-adding to outputs
2024-10-27 17:58:54.905 (1323) [Schedule] /opt/fpp/src/Scheduler.cpp:729: Current Schedule: (Status: '+' = Enabled, '-' = Disabled, '!' = Outside Date Range, '*' = Repeat, Stop (G)raceful/(L)oop/(H)ard
2024-10-27 17:58:54.905 (1323) [Schedule] /opt/fpp/src/Scheduler.cpp:730: Stat Start & End Dates       Days          Start & End Times   Playlist/Command
2024-10-27 17:58:54.905 (1323) [Schedule] /opt/fpp/src/Scheduler.cpp:731: ---- ----------------------- ------------- ------------------- ---------------------------------------------
2024-10-27 17:58:54.905 (1323) [Schedule] /opt/fpp/src/Scheduler.cpp:734: + *G 2022-11-25 - 2028-12-31 Everyday      SunSet   - 22:00:00 Christmas_Music
2024-10-27 17:58:54.905 (1323) [Schedule] /opt/fpp/src/Scheduler.cpp:734: - *G 2022-12-07 - 2027-12-31 Everyday      22:01:00 - 22:10:00 Christmas Countdown
2024-10-27 17:58:54.905 (1323) [Schedule] /opt/fpp/src/Scheduler.cpp:734: +  G 2023-12-06 - 2028-12-31 Everyday      22:00:00            Remote Falcon - Turn Viewer Control Off
2024-10-27 17:58:54.905 (1323) [Schedule] /opt/fpp/src/Scheduler.cpp:734: +  G 2022-11-25 - 2028-12-31 Everyday      SunSet              Remote Falcon - Update Remote Playlist
2024-10-27 17:58:54.905 (1323) [Schedule] /opt/fpp/src/Scheduler.cpp:734: +  G 2022-11-25 - 2028-12-31 Everyday      SunSet              Remote Falcon - Turn Viewer Control On
2024-10-27 17:59:11.708 (2126) [General] /opt/fpp/src/fppd.cpp:1077: Publishing statistics because of "Startup" = {"status":"OK","uuid":"M1-1000000010ab5250"}
2024-10-27 18:58:47.114 (1323) [ChannelOut] /opt/fpp/src/channeloutput/UDPOutput.cpp:349: UDP Interface eth0 now up
2024-10-27 19:05:30.791 (1323) [General] /opt/fpp/src/fppd.cpp:803: Shutting down main loop.
2024-10-27 19:05:30.792 (1323) [General] /opt/fpp/src/fppd.cpp:1025: Stopping channel output thread.
2024-10-27 19:05:30.892 (1323) [General] /opt/fpp/src/fppd.cpp:1029: Main Loop complete, shutting down.
2024-10-27 19:05:31.905 (1323) [General] /opt/fpp/src/fppd.cpp:786: Performing Restart.
2024-10-27 19:05:32.269 (1323) [General] /opt/fpp/src/log.cpp:353: =========================================
2024-10-27 19:05:32.269 (1323) [General] /opt/fpp/src/log.cpp:354: FPP 8.2
2024-10-27 19:05:32.269 (1323) [General] /opt/fpp/src/log.cpp:355: Branch: v8.2
2024-10-27 19:05:32.269 (1323) [General] /opt/fpp/src/log.cpp:367: Model: Raspberry Pi 4 Model B Rev 1.2
2024-10-27 19:05:32.269 (1323) [General] /opt/fpp/src/log.cpp:371: =========================================
2024-10-27 19:05:32.271 (1323) [General] /opt/fpp/src/fppd.cpp:661: Creating MQTT
2024-10-27 19:05:32.271 (1323) [Control] /opt/fpp/src/mqtt.cpp:175: Using MQTT client id of FPP-Main_7773 
2024-10-27 19:05:32.271 (1323) [Control] /opt/fpp/src/mqtt.cpp:198: No CA File specified for MQTT
2024-10-27 19:05:32.271 (1323) [General] /opt/fpp/src/fppd.cpp:709: Creating Scheduler, Playlist, and Sequence
2024-10-27 19:05:32.280 (17695) [Control] /opt/fpp/src/mqtt.cpp:350: Mosquitto Connected
2024-10-27 19:05:32.280 (17695) [Control] /opt/fpp/src/mqtt.cpp:352: HandleConnect() called before can process messages.  Won't Register topics yet.
2024-10-27 19:05:32.282 (1323) [Schedule] /opt/fpp/src/Scheduler.cpp:729: Current Schedule: (Status: '+' = Enabled, '-' = Disabled, '!' = Outside Date Range, '*' = Repeat, Stop (G)raceful/(L)oop/(H)ard
2024-10-27 19:05:32.282 (1323) [Schedule] /opt/fpp/src/Scheduler.cpp:730: Stat Start & End Dates       Days          Start & End Times   Playlist/Command
2024-10-27 19:05:32.282 (1323) [Schedule] /opt/fpp/src/Scheduler.cpp:731: ---- ----------------------- ------------- ------------------- ---------------------------------------------
2024-10-27 19:05:32.282 (1323) [Schedule] /opt/fpp/src/Scheduler.cpp:734: + *G 2022-11-25 - 2028-12-31 Everyday      SunSet   - 22:00:00 Christmas_Music
2024-10-27 19:05:32.282 (1323) [Schedule] /opt/fpp/src/Scheduler.cpp:734: - *G 2022-12-07 - 2027-12-31 Everyday      22:01:00 - 22:10:00 Christmas Countdown
2024-10-27 19:05:32.282 (1323) [Schedule] /opt/fpp/src/Scheduler.cpp:734: +  G 2023-12-06 - 2028-12-31 Everyday      22:00:00            Remote Falcon - Turn Viewer Control Off
2024-10-27 19:05:32.282 (1323) [Schedule] /opt/fpp/src/Scheduler.cpp:734: +  G 2022-11-25 - 2028-12-31 Everyday      SunSet              Remote Falcon - Update Remote Playlist
2024-10-27 19:05:32.282 (1323) [Schedule] /opt/fpp/src/Scheduler.cpp:734: +  G 2022-11-25 - 2028-12-31 Everyday      SunSet              Remote Falcon - Turn Viewer Control On
2024-10-27 19:05:32.298 (1323) [General] /opt/fpp/src/fppd.cpp:712: Creation of Scheduler, Playlist, and Sequence Complete
2024-10-27 19:05:32.337 (1323) [ChannelOut] /opt/fpp/src/channeloutput/ChannelOutputSetup.cpp:321: UDPOutput:  Determined range needed 0 - 1530
2024-10-27 19:05:32.337 (1323) [ChannelOut] /opt/fpp/src/channeloutput/ChannelOutputSetup.cpp:321: UDPOutput:  Determined range needed 1530 - 27030
2024-10-27 19:05:32.338 (1323) [ChannelOut] /opt/fpp/src/channeloutput/ChannelOutputSetup.cpp:351: Determined range needed 0 - 27030
2024-10-27 19:05:32.339 (1323) [Command] /opt/fpp/src/command.cpp:70: Initializing Command Module
2024-10-27 19:05:32.339 (1323) [E131Bridge] /opt/fpp/src/e131bridge.cpp:316: Universe Count = 0
2024-10-27 19:05:32.553 (1323) [Control] /opt/fpp/src/mqtt.cpp:335: Mosquitto SetReady()
2024-10-27 19:05:32.553 (1323) [Control] /opt/fpp/src/mqtt.cpp:350: Mosquitto Connected
2024-10-27 19:05:32.554 (1323) [Control] /opt/fpp/src/mqtt.cpp:356: Mosquitto Connected.... Will Subscribe to Topics
2024-10-27 19:05:32.554 (1323) [Control] /opt/fpp/src/mqtt.cpp:382: MQTT HandleConnect Complete
2024-10-27 19:05:32.554 (1323) [General] /opt/fpp/src/fppd.cpp:894: Starting main processing loop
2024-10-27 19:05:35.554 (17722) [Control] /opt/fpp/src/Events.cpp:218: Starting Publish Thread
2024-10-27 19:06:13.739 (18013) [General] /opt/fpp/src/fppd.cpp:1077: Publishing statistics because of "Startup" = {"status":"OK","uuid":"M1-1000000010ab5250"}
```

**Comment by ghormann** [comment]
If you want this investigated, I'd suggest opening a new ticket.   That said, I have the feeling your wireless is not fully up when fppd starts and that adding fppd boot delay (Settings->System) is probably your best solution.    The reason I believe it is a networking delay is 
* the wireless bridge is on the same power supply and not always on
* The error "Error, unable to connect to Mosquitto Broker at [10.0.0.243](http://10.0.0.243/): 14" suggest a network error vs time
* The error "Could not ping host 10.0.0.205"  (also 206) suggest that these devices are not accessible via the network yet but 6 seconds later, these addresses are accessible.   

Bottom line, adding some delay to fppd starting would allow time for the network to stabilize
